### PR TITLE
NAS-130002 / 24.10 / Replace TrueNAS with vendor in installer

### PR DESCRIPTION
**Pull request by aiden3c**
This replaces all written instances of TrueNAS with the vendor information, defaulting to TrueNAS if no vendor is supplied

### CODE REVIEW

File: truenas_installer/installer.py [hunk 1]
```diff
@@ -2,7 +2,8 @@
 
 
 class Installer:
-    def __init__(self, version, dmi):
+    def __init__(self, version, dmi, vendor=None):
```
Comment (themylogin) [inline truenas_installer/installer.py:5]: Why the default value?

Comment (themylogin) [inline truenas_installer/installer.py:5]: This default value will never be used
```suggestion
    def __init__(self, version, dmi, vendor):
```

Comment (aiden3c) [reply truenas_installer/installer.py:5]: Good catch

File: truenas_installer/__main__.py [hunk 2]
```diff
@@ -20,9 +21,15 @@ def main():
     with open("/etc/version") as f:
         version = f.read().strip()
 
+    try:
```
Comment (yocalebo) [inline truenas_installer/__main__.py:24]: Let's be extra safe here. I'd rather just catch all exceptions and pass.

```
vendor = "TrueNAS"
try:
    with open(...
        vendor = ...
except Exception:
    pass

**Review by yocalebo — CHANGES_REQUESTED** [review]

**Review by yocalebo — APPROVED** [review]
Thanks!

**Pull request merged**